clang/lib/Sema/SemaStmtAttr.cpp

```diff
@@ -625,6 +625,43 @@ static Attr *handleHLSLControlFlowHint(Sema &S, Stmt *St, const ParsedAttr &A,
   return ::new (S.Context) HLSLControlFlowHintAttr(S.Context, A);
 }
 
+static Attr *handleAtomicAttr(Sema &S, Stmt *St, const ParsedAttr &A,
+                              SourceRange Range) {
+  if (!isa<CompoundStmt>(St)) {
```

Comment:
Does this ever get hit?  The infrastructure from Attr.td should handle this.

Reply:
Yes we still need this. The infrastructure from Attr.td parses the attribute and creates ParsedAttr. The AtomicAttr in AST still needs to be created in Sema by this handler. However, we could avoid the specific parsing function for AtomicAttr in ParseDecl.cpp

Reply:
I mean the `CompoundStmt` check.  The `Subject` in the `Attr.td` should have already prevent anything wiht a `CompoundStmt` Statement from getting here.

Reply:
You are right. The CompoundStmt check is not necessary since equivalent check is done in Parser. Will remove this check here.